lib/matplotlib/legend.py

```diff
@@ -565,19 +565,23 @@ def get_legend_handler(legend_handler_map, orig_handle):
         method-resolution-order. If no matching key is found, it
         returns None.
         """
-        legend_handler_keys = list(six.iterkeys(legend_handler_map))
-        if orig_handle in legend_handler_keys:
-            handler = legend_handler_map[orig_handle]
+        try:
+            hash(orig_handle)
```

Comment:
in `cbook.py` we have a `ishashable` function which basically does this, but returns a boolean.  I can see going either way on using it here.